pkg/client/monitoring/v1alpha1/types.go

```diff
@@ -80,6 +80,8 @@ type PrometheusSpec struct {
 	RoutePrefix string `json:"routePrefix,omitempty"`
 	// Storage spec to specify how storage shall be used.
 	Storage *StorageSpec `json:"storage,omitempty"`
+	// Pvc A pvc spec to be used by the Prometheus statefulsets.
+	Pvc *v1.PersistentVolumeClaim `json:"pvc,omitempty"`
```

Comment:
You mean a `StatefulSet.Spec.VolumeClaimTemplates` ?
I think it is confusing, because VolumeClaimTemplates is a list, and it will suggest that it supports multiple volumes, aside that, thats my only argument about using VolumeClaimTemplates.

Adding it to a field in the Storage Spec, is a good idea.